File: task-03/Javascript/3/Sub-Task3.js
function printDiamond(n) {
    for (let i = 0; i < n; i++) {
        let spaces = ' '.repeat(n - i - 1);
        let stars = '*'.repeat(2 * i + 1);
        console.log(spaces + stars);
    }

    for (let i = n - 2; i >= 0; i--) {
        let spaces = ' '.repeat(n - i - 1);
        let stars = '*'.repeat(2 * i + 1);
        console.log(spaces + stars);
    }
}

const readline = require('readline');
const rl = readline.createInterface({
    input: process.stdin,
    output: process.stdout
});

rl.question('Enter a number: ', (input) => {
    const n = parseInt(input, 10);
    if (isNaN(n) || n <= 0) {
        console.log('Please enter a valid positive number.');
    } else {
        printDiamond(n);
    }
    rl.close();
});

File: task-03/Rust/3/Sub-Task3.rs
use std::io;

fn main() {
    let mut input = String::new();
    println!("Enter a number:");
    io::stdin().read_line(&mut input).expect("Failed to read line");
    let n: usize = input.trim().parse().expect("Please enter a valid number");

    print_diamond(n);
}

fn print_diamond(n: usize) {
    for i in 0..n {
        for _ in 0..(n - i - 1) {
            print!(" ");
        }
        for _ in 0..(2 * i + 1) {
            print!("*");
        }
        println!();
    }

    for i in (0..n - 1).rev() {
        for _ in 0..(n - i - 1) {
            print!(" ");
        }
        for _ in 0..(2 * i + 1) {
            print!("*");
        }
        println!();
    }
}
